libs/barista-components/overlay/src/overlay-trigger.ts

```diff
@@ -84,6 +86,17 @@ export class DtOverlayTrigger<T>
     this._config = value;
   }
 
+  /** Emits on every pinned state change */
+  @Output()
+  pinnedChanged = new EventEmitter<boolean>();
+
+  /** Whether the underlying overlay is pinned or not */
+  get isOverlayPinned(): boolean | undefined {
+    return this._dtOverlayRef?.pinned;
+  }
+
+  private _pinnedChangedSubscribtion: Subscription;
```

Comment:
```suggestion
  private _pinnedChangedSubscription = EMPTY.subscribe();
```
You can use an empty subscription as the initial value so that you can safely unsubscribe even if the real subscription is never created.